Configurations/windows-makefile.tmpl

```diff
@@ -3,14 +3,16 @@
 ##
 ## {- join("\n## ", @autowarntext) -}
 {-
+
+ our $lib_suffix = $target{lib_suffix} || "";
  our $objext = $target{obj_extension} || ".obj";
  our $resext = $target{res_extension} || ".res";
  our $depext = $target{dep_extension} || ".d";
  our $exeext = $target{exe_extension} || ".exe";
- our $libext = $target{lib_extension} || ".lib";
- our $shlibext = $target{shared_extension} || ".dll";
- our $shlibextimport = $target{shared_import_extension} || ".lib";
- our $dsoext = $target{dso_extension} || ".dll";
+ our $libext = $lib_suffix . ($target{lib_extension} || ".lib");
+ our $shlibext = $lib_suffix . ($target{shared_extension} || ".dll");
+ our $shlibextimport = $lib_suffix . ($target{shared_import_extension} || ".lib");
+ our $dsoext = $lib_suffix . ($target{dso_extension} || ".dll");
```

Comment:
I haven't tested this, but if this line means that even engine dlls will get addendum in its name, then one has to recognize that when you ask to load engine X, the name is composed according to platform-specific predefined rules, so there is no place for name customization...

Reply:
@dot-asm Could you elaborate a bit more on your concern? I am not familiar with your build chain and I have not messed with library names before. I this about the argument you need to pass to `dlopen()`? Is this [patch](https://www.npcglib.org/~stathis/downloads/openssl-1.1.0f.patch) going a better way? Please be a bit more explicit about what needs to be changed.

I am not sure what you want to say with the following statement
_So it's not like_ **the** / **a** _suggestion **(** is **)** to take some code from unix-Makefile.tmpl_ Depending on the missing word the meaning changes dramatically. Judging by the following sentence the first variant is more likely and you want the `$lib_suffix` to be more documented?! I think this can only be done if the technical part of this PR is set. Should it be put in the unix makefile template or not. How will the modifications work on Windows eventually?

Reply:
Engine load request is meant to be OS-neutral. In sense that you'll use same call on all OS, and libcrypto till take care of OS specifics. If you ask load engine X, on Windows it will attempt to load X.DLL. If you produce XMD.DLL, then attempt to load X is bound to fail, which would be inappropriate.

> what you want to say

Right, I'm not in good terms with English, especially articles :-) What I suggest is to re-use `shlib_variant` attribute name (introduced earlier for similar kind of thing). I didn't have close look and rather assumed that taking snippets straight from unix-Makefile.tmpl wouldn't work. Richard seems to come to different conclusion, so this ought to mean that he looked closer. But taken code from unix-Makefile.tmpl or not, it would be only appropriate to add few words to Configurations/README about that `shlib_variant` is significant even on Windows and what does do there.